internal/appsec/waf.go

```diff
@@ -235,32 +235,32 @@ func newHTTPWAFEventListener(handle *wafHandle, addresses map[string]struct{}, t
 		op.On(httpsec.OnHandlerOperationFinish(func(op *httpsec.Operation, res httpsec.HandlerOperationRes) {
 			defer wafCtx.Close()
 
-			values := make(map[string]interface{}, 1)
+			values := map[string]any{}
 			if _, ok := addresses[serverResponseStatusAddr]; ok {
-				values[serverResponseStatusAddr] = res.Status
+				// serverResponseStatusAddr is a string address, so we must format the status code...
+				values[serverResponseStatusAddr] = fmt.Sprintf("%d", res.Status)
 			}
 
 			// Run the WAF, ignoring the returned actions - if any - since blocking after the request handler's
 			// response is not supported at the moment.
-			matches, _ := runWAF(wafCtx, values, timeout)
+			wafResult := runWAF(wafCtx, waf.RunAddressData{Persistent: values}, timeout)
 
 			// Add WAF metrics.
-			rInfo := handle.RulesetInfo()
+			wafDiags := handle.Diagnostics()
```

Comment:
To avoid copying the diagnostic struct with this call on each request, can you please move it to the parent closure where the handle is (the top-level one actually) to prepare it only once per handle and simply reference it from here? Or make it a cached struct in go-libddwaf.

(The beauty of dyngo's nested closures to control the granularity of data inits :-)).

Reply:
It's already a cached struct... I'm actually unsure why this one is exposed by a method instead of the field just being public...?

https://github.com/DataDog/go-libddwaf/blob/f4a695824ff1d930fcb3382bd8e0371b78664b38/handle.go#L91-L94